### PR TITLE
Remove work pipelining

### PR DESCRIPTION
**Pull request by jjyao**
## Why are these changes needed?
This is a follow-up of https://github.com/ray-project/ray/pull/21878. This is a feature that's not enabled so removing it to make the codebase simpler.

Note: this removes a config so it might affect people.

## Checks

- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

**Review by scv119 — APPROVED** [review]

**Review by ericl — APPROVED** [review]

File: src/ray/common/ray_config_def.h [hunk 1]
```diff
@@ -326,11 +326,6 @@ RAY_CONFIG(int64_t, enable_metrics_collection, true)
 // Max number bytes of inlined objects in a task rpc request/response.
 RAY_CONFIG(int64_t, task_rpc_inlined_bytes_limit, 10 * 1024 * 1024)
 
-/// Maximum number of tasks that can be in flight between an owner and a worker for which
-/// the owner has been granted a lease. A value >1 is used when we want to enable
-/// pipelining task submission.
-RAY_CONFIG(uint32_t, max_tasks_in_flight_per_worker, 1)
```
Comment (scv119) [inline src/ray/common/ray_config_def.h:332]: maybe leave it and mark it deprecated and get rid of it in next 1-2 releases

Comment (ericl) [reply src/ray/common/ray_config_def.h:332]: I don't think this is necessary--- all Ray internal flags are considered private unless otherwise documented.

Comment (jjyao) [reply src/ray/common/ray_config_def.h:332]: Could you give me an example of public flag? I'm curious how we document it so that users know it's public

Comment (ericl) [reply src/ray/common/ray_config_def.h:332]: It should be pretty obvious, e.g., have all caps version and a comment.

**Comment by jjyao** [comment]
It's ready to be merged.

**Pull request merged**